lib/capybara/server/middleware.rb

```diff
@@ -7,16 +7,16 @@ class Counter
         attr_reader :value
 
         def initialize
-          @value = 0
+          @value = []
           @mutex = Mutex.new
         end
 
-        def increment
-          @mutex.synchronize { @value += 1 }
+        def increment(env)
+          @mutex.synchronize { @value.push(env) }
         end
 
-        def decrement
-          @mutex.synchronize { @value -= 1 }
+        def decrement(env)
+          @mutex.synchronize { @value.delete_at(@value.index(env)) }
```

Comment:
Usually, it's good practice to provide a fallback when `@value.index(env)` does not find an element. However, since we are using `@counter` privately and in a sequential increment => decrement manner, I do not think this necessitates a fallback. 

Let me know what you think.

Reply:
Yes please -- `@value.delete_at(@value.index(env) || @value.length)` just for safety